aiida_quantumespresso/calculations/pw.py

```diff
@@ -152,6 +154,26 @@ def define(cls, spec):
                     'is `False` and/or `electron_maxstep` is 0.')
         # yapf: enable
 
+    @staticmethod
+    def validate_inputs(value, _):
+        """Validate the top level namespace.
+
+        - Check that the restart input parameters are set when a ``parent_folder`` is provided.
+        """
+        if 'parent_folder' in value:
+
+            parameters = value['parameters'].get_dict()
+            if (
+                parameters.get('CONTROL', {}).get('restart_mode', None) != 'restart' and
+                parameters.get('ELECTRONS', {}).get('startingpot', None) != 'file' and
+                parameters.get('ELECTRONS', {}).get('startingwfc', None) != 'file'
+            ):
```

Comment:
Just briefly commenting that we may want to also include `parameters['CONTROL']['calculation'] not in {'nscf', 'bands'}` as one of the possibilities. From the [PW input web](https://www.quantum-espresso.org/Doc/INPUT_PW.html#idm883): 

> startingpot = 'file'
> 
> (...) For nscf and bands calculation this is the default
> and the only sensible possibility.

Reply:
@sphuber the default for `restart_mode` is `'from_scratch'`, which is the required setting when using `startingpot` or `startingwfc`. So only setting `ELECTRONS.startingpot == 'file'` works perfectly fine, I think.

However, @ramirezfranciscof rightly points out that not setting _any_ inputs when running `nscf` or `bands` calculations is also fine, so we shouldn't warn in these cases. Maybe the validation here should be to make sure that the `parent_folder` is actually there, and a non-sensible value for the restart inputs is not set?

Reply:
Would it be sensible to enforce that a `parent_folder` is provided for a `nscf` or `bands` calculation?

Reply:
Similarly, since running with `ELECTRONS.startingpot == 'file'` is the only sensible choice for these calculations, should we override the inputs and warn that we have done so in case they are set incorrectly?

Reply:
> I'm a little apprehensive about doing this though. But maybe just raising warnings here isn't enough?

One idea that I had been thinking about lately was to add some sort of input keyword for workchains like `force` (`failsafe = False`?) that allows the user to turn some errors into warnings so that they can fully mess with the workchain taking responsability of the results. Not sure if this is the place to discuss this as a more general approach, but your comment reminded me of this.

Reply:
> Would it be sensible to enforce that a parent_folder is provided for a nscf or bands calculation?

I think this is necessary no? Does it ever make sense to run either of these without a `parent_folder`?

> should we override the inputs and warn that we have done so in case they are set incorrectly? I'm a little apprehensive about doing this though. But maybe just raising warnings here isn't enough?

I would say that if we _know_ that for certain calculation types certain inputs _have_ to be set, than we should do that. If there are settings that are likely incorrect, but could still make sense in certain edge cases, then a warning might be better.

I think there is still a problem with the conditional though. If `calculation == 'scf'` _and_ there is a `parent_folder` you _only_ warn if _all_ three parameters are incorrectly set (you are using `and`). But shouldn't this be `or` statements? If _any_ of the parameters are incorrect, it should warn?

Reply:
> I would say that if we know that for certain calculation types certain inputs have to be set, than we should do that. If there are settings that are likely incorrect, but could still make sense in certain edge cases, then a warning might be better.

Again, I would be in favor of just raising errors for non-sensible inputs instead of just overriding them. If the user somehow has set wrong inputs, this might indicate a problem in their workflow they would want to be aware of, instead of having it silently fixed in the work chain. In this light, I may change the warnings for `nscf` and `nbands` to actually raise errors. If there _is_ some edge case that we missed, we can always revert this later with a warning that clearly explains this use case.

> I think there is still a problem with the conditional though. If calculation == 'scf' and there is a parent_folder you only warn if all three parameters are incorrectly set (you are using and). But shouldn't this be or statements? If any of the parameters are incorrect, it should warn?

I think the logic is inverted (sort of): In case `parent_folder` is set, _at least_ one of `restart_mode == 'restart'`, `startingpot == 'file'` or `startingwfc == 'file'` must be set for an `scf` calculation. Perhaps it would be clearer to change:

```Python
if (
    parameters.get('CONTROL', {}).get('CONTROL', None) != 'restart' and
    parameters.get('ELECTRONS', {}).get('startingpot', None) != 'file' and
    parameters.get('ELECTRONS', {}).get('startingwfc', None) != 'file'
):
```

into 

```Python
if not any([
    parameters.get('CONTROL', {}).get('CONTROL', None) == 'restart' and
    parameters.get('ELECTRONS', {}).get('startingpot', None) == 'file' and
    parameters.get('ELECTRONS', {}).get('startingwfc', None) == 'file'
]):
```

We should consider if there are other reasons to provide a `parent_folder`, though. 🤔

Reply:
> One idea that I had been thinking about lately was to add some sort of input keyword for workchains like force (failsafe = False?) that allows the user to turn some errors into warnings so that they can fully mess with the workchain taking responsability of the results. Not sure if this is the place to discuss this as a more general approach, but your comment reminded me of this.

@sphuber is there a way to disable _all_ validation? As long as we carefully stick to using validators to make sure inputs are sensible (and not just override them), we can disable validation and the user can go where there be dragons. ^^

Reply:
No, currently there is no way to do this dynamically through the inputs. Might be worth a requested feature discussion on `aiida-core`, but it may have only niche use